### PR TITLE
Stairsplottype

### PR DESCRIPTION
**Pull request by kkoutris**
## PR Summary
Added new plot type(stairs) to the documentation as asked by #23204.
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
Add stairs plot type to the documentation as asked from #23204

### CODE REVIEW

File: examples/text_labels_and_annotations/angle_annotation.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def update_text(self):
         r = s / 2
         if self.textposition == "inside":
             r = s / np.interp(angle_span, [60, 90, 135, 180],
-                                          [3.3, 3.5, 3.8, 4])
+                                          [10, 3.5, 11.4, 4])
```
Comment (jklymak) [inline examples/text_labels_and_annotations/angle_annotation.py:188]: Unrelated change?

File: plot_types/basic/stairs.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"""
+==========
+stairs(x, y)
+==========
```
Comment (jklymak) [inline plot_types/basic/stairs.py:4]: Docs won't build if formatting is incorrect: `/home/circleci/project/doc/plot_types/basic/stairs.rst:24: WARNING: Title overline too short.`
```suggestion
============
stairs(x, y)
============
```

File: plot_types/basic/stairs.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# plot
+fig, ax = plt.subplots()
+
+ax.stairs(x, y, orientation='horizontal')
+
+ax.set(xlim=(0, 8), xticks=np.arange(1, 8),
+       ylim=(0, 8), yticks=np.arange(1, 8))
+
+plt.show()
```
Comment (jklymak) [inline plot_types/basic/stairs.py:26]: We need the extra line for our linting.

**Review by jklymak — CHANGES_REQUESTED** [review]
Thanks for this!

OTOH I don't think we should have both `stairs` and `steps`.  This should also be accompanied by a change to the cheatsheet if the `stairs` is really the preferred grammar here.

**Comment by jklymak** [comment]
I'll move to draft for now - feel free to move to ready when you have a chance to address..

**Comment by QuLogic** [comment]
I think this has been replaced by #26233. Thank you for your effort @kkoutris; we're sorry that this was not accepted, but we hope to hear from you again.

**Pull request closed without merging**